public/app/plugins/datasource/prometheus/datasource.ts

```diff
@@ -493,8 +482,12 @@ export class PrometheusDatasource extends DataSourceApi<PromQuery, PromOptions>
       }
     }
 
-    return this._request(url, data, { requestId: query.requestId, headers: query.headers }).pipe(
-      catchError(err => {
+    // This can return a scalar data as well
```

Comment:
I am not sure I understand the meaning of this comment. If this can return a scalar data as well, shouldn't the typing reflect that.